### ISSUE TITLE
Crash in `parse_types_from_source` with options `-x c` and `--std c99`

### ISSUE THREAD

**Issue by ElykDeer** [Vector35/binaryninja-api#6225]
There appears to be a bug in Binary Ninja's type parser when combining -x c and --std c99 options. The segfault occurs specifically when both flags are present, while using either flag alone works (though with expected parsing errors).

1. Segfault only happens with both -x c and --std c99
2. Issue reproduces across platforms (arm32 and x86-64)
3. BN version: 4.3.6468-dev

(credit to @fosdickio for the report)

**Comment by CouleeApps** [comment]
Can't reproduce? What steps are you taking to cause this crash and what type / args are you parsing that causes this?

**Comment by waskyo** [comment]
Hi, attached is the source file that is causing the issue. I'm able to parse other files without problems.

Here are my repro steps:
```python
from binaryninja import *
bv = load("/bin/ls") # This is x86-64, but i've also tried arm32 with the same results
with open("/tmp/segfault.c") as f:
     source = f.read()

tp = TypeParser.default
tp.parse_types_from_source(source=source, file_name='segfault.c', platform=bv.platform, options=["-x", "c", "--std", "c99"])
Segmentation fault
```

[segfault.c.txt](https://github.com/user-attachments/files/18009434/segfault.c.txt)

**Comment by CouleeApps** [comment]
After playing around with this, I have managed to reproduce the crash. The stack trace points within clang, handling function type attributes. I'm betting it's something in the custom type attribute parsing patches I wrote, considering I wouldn't expect clang to crash on its own. I will investigate further.

**Comment by CouleeApps** [comment]
Fixed with an additional null check in the type attribute handling patches we have for clang. Will be fixed in >=4.3.6587
